src/main/export/exportToMP4.ts

```diff
@@ -0,0 +1,137 @@
+import { BrowserWindow } from 'electron';
+import { unlink } from 'fs/promises';
+import path, { join } from 'path';
+import ffmpeg from 'fluent-ffmpeg';
+import { RuntimeProject, Cut } from '../../sharedTypes';
+import { mkdir } from '../util';
+import convertTranscriptToCuts from '../../transcriptProcessing/transcriptToCuts';
+import { ffmpegPath } from '../ffUtils';
+
+const createTempCutVideo: (
+  inputPath: string,
+  outputDir: string,
+  startTime: number,
+  duration: number,
+  outputNumber: number
+) => Promise<string> = (
+  inputPath,
+  outputDir,
+  startTime,
+  duration,
+  outputNumber
+) => {
+  const outputPath = join(outputDir, `/cut${outputNumber}.mp4`);
+
+  return new Promise((resolve, reject) => {
+    ffmpeg()
+      .input(inputPath)
+      .setStartTime(startTime)
+      .setDuration(duration)
+      .output(outputPath)
+      .on('end', (stdout: string, stderr: string) => {
+        if (stdout) console.log(`FFMPEG stdout: ${stdout}`);
+        if (stderr) console.error(`FFMPEG stderr: ${stderr}`);
+
+        console.log('cut video success');
+        resolve(outputPath);
+      })
+      .on('error', (stdout: string, stderr: string) => {
+        if (stdout) console.log(`FFMPEG stdout: ${stdout}`);
+        if (stderr) console.error(`FFMPEG stderr: ${stderr}`);
+
+        console.log('cut video error');
+        reject(stderr);
+      })
+      .run();
+  });
+};
+
+const allTempCutAllVideos: (
+  cuts: Cut[],
+  source: string | null,
+  tempFileDir: string
+) => Promise<string>[] = (cuts, source, tempFileDir) => {
+  return cuts.map((cut, idx) => {
+    return createTempCutVideo(
+      source ?? 'FILE PATH NOT FOUND',
+      tempFileDir,
+      cut.startTime,
+      cut.duration,
+      idx
+    );
+  });
+};
+
+const mergeTempCutVideos: (
+  inputPaths: string[],
+  outputPath: string
+) => Promise<boolean> = (inputPaths, outputPath) => {
+  const mergedTempCut = ffmpeg();
+
+  inputPaths.forEach((inputPath) => {
+    mergedTempCut.mergeAdd(inputPath);
+  });
+
+  return new Promise((resolve, reject) => {
+    mergedTempCut
+      .mergeToFile(outputPath)
+      .on('end', (stdout: string, stderr: string) => {
+        if (stdout) console.log(`FFMPEG stdout: ${stdout}`);
+        if (stderr) console.error(`FFMPEG stderr: ${stderr}`);
+
+        resolve(true);
+      })
+      .on('error', (stdout: string, stderr: string) => {
+        if (stdout) console.log(`FFMPEG stdout: ${stdout}`);
+        if (stderr) console.error(`FFMPEG stderr: ${stderr}`);
+
+        reject(stderr);
+      });
+  });
+};
+
+const deleteTempCutVideos: (tempCutVideoPaths: string[]) => Promise<void[]> = (
```

Comment:
Change return type to `Promise<void>`

Reply:
I'm using `Promise.all()` here, and it returns `Promise<void[]>`

Reply:
Ah if you do `await Promise.all(…)` instead then it will return Promise<void>. You’re not using the return array so that makes more sense I feel

Reply:
Ah if you do `await Promise.all(…)` instead then it will return Promise<void>. You’re not using the return array so that makes more sense I feel